.github/workflows/main.yml

```diff
@@ -0,0 +1,125 @@
+on:
+  pull_request:
+  push:
+
+jobs:
+  build:
+    runs-on: Ubuntu-latest
+    strategy:
+      matrix:
+        node-version:
+        - 14.x
+
+    steps:
+    - name: Fetch the src
+      uses: actions/checkout@v2
+    - name: >-
+        Set up NodeJS ${{ matrix.node-version }}
+        with the global NPM registry
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+        registry-url: https://registry.npmjs.org
+
+    - name: Build a package tarball artifact
+      run: npm pack
+
+    - name: Save the package tarball as a GHA artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: npm-package-tarball
+        path: >-
+          ansible-ansible-language-server-*.tgz
+
+  test:
+    needs:
+    - build
+
+    runs-on: ${{ matrix.os }}-latest
+
+    strategy:
+      matrix:
+        node-version:
+        - 16.x
+        - 14.x
```

Comment:
Any particular reason to add an older node version in the matrix? Can't this be the same version as the extension node version https://github.com/ansible/vscode-ansible/blob/main/package.json#L335

Reply:
There's no compelling reason not to support LTS versions of node. What if somebody implementing a Vim integration would use those? FWIW I wouldn't worry about it for as long as it doesn't cause problems.

Reply:
We will support and use only one version of node, newest one that is available on our production jenkins, GHA and fedora stable. "16" is fine now. 

We do not have the bandwidth to support more, any developer should be able to setup his local env. Less variance will translate to less wasted time debugging build failures.

Reply:
If you do not have further objections, please mark this thread as resolved. TIA!

Reply:
@ssbarnea let's drop it once it causes actual non-hypothetical bug reports.

Reply:
Sorry, but I am not willing to let this in. It does bother me considerably for multiple reasons, one of them being that I need to scroll among a plethora of duplicated CI jobs. Basically we have 8x more jobs that we need: 4 variants of node and two push/pull-request.

Anyone can compare how checks look at #18 (28 jobs) versus #17 which has only 3 checks, one for each platform.

More jobs/check does not necessary means better tested code, or easier to debug. It adds noise and extra delays.

Our goal is not to test if different version of build tools (node) work the same, is only to build/test the package, not the tools. On vscode-ansible we already require specific version of node and prevent execution with unsupported versions.

What prevents us from using a similar pipeline setup?

Reply:
> Sorry, but I am not willing to let this in. It does bother me considerably for multiple reasons, one of them being that I need to scroll among a plethora of duplicated CI jobs.

Well, this sounds pretty much like a personal preference rather than measurable data. I appreciate the sentiment but I like seeing all the distinct checks in the PRs. And they are not annoying because most of the time the reviews happen in the Files tab with those statuses not visible. Still, it's a great benefit to be able to quickly reach and inspect exact envs and places that fail.

If you have improvement ideas, I suggest you file an issue in Refined GitHub to see if anybody else shares your frustration. It is a great browser extension and I'm sure they'd be happy to improve the UI for the users if the requested change is relatable enough.

> Basically we have 8x more jobs that we need: 4 variants of node and two push/pull-request.

This is a separate thing to discuss as it relates to the workflow we currently encourage. I detailed the reason here https://github.com/ansible/ansible-language-server/pull/18#discussion_r709876416. If we, as a group, want to start filtering the events as you suggest, we'd have to agree to change our workflow which shouldn't be a blocker of this PR.

> Anyone can compare how checks look at #18 (28 jobs) versus #17 which has only 3 checks, one for each platform.

https://github.com/ansible/ansible-language-server/pull/18#discussion_r709876416

> More jobs/check does not necessary means better tested code, or easier to debug. It adds noise and extra delays.
> 
> Our goal is not to test if different version of build tools (node) work the same, is only to build/test the package, not the tools. On vscode-ansible we already require specific version of node and prevent execution with unsupported versions.
> 
> What prevents us from using a similar pipeline setup?

Well, as I explained above, "noise" is subjective so there's nothing to add except it's not noisy, actually :man_shrugging:
I agree with the statement that a matrix does not necessarily mean better tests. However, it does have the potential to catch regressions once we have proper testing and that's what I'm going for.
As for the unsupported versions, I think you yourself have found an argument against your request earlier today, in Slack — Theia uses NodeJS v12 as we've established. I'll drop v10 tho.